frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx

```diff
@@ -185,12 +185,37 @@ export function ObjectDetailView({
     if (!_.isEmpty(tableForeignKeys)) {
       loadFKReferences();
     }
-    window.addEventListener("keydown", onKeyDown, true);
+  });
 
-    return () => {
-      window.removeEventListener("keydown", onKeyDown, true);
+  useEffect(() => {
+    if (hasNotFoundError) {
+      return;
+    }
+
+    const onKeyDown = (event: KeyboardEvent) => {
```

Comment:
Moves `onKeyDown` to the effect to avoid deps issues.

Reply:
@uladzimirdev If a callback is only used inside the effect itself I think it's too verbose to create it outside the effect with `useCallback`.

Putting that function inside the effect would not only eliminate the need to have it in the dependency array, but also limit the scope of the callback function so its potential of being misused/misunderstood is a lot less, since you can only access the function inside the `useEffect` you will know it will only be used here. In contrast, when putting this callback outside the effect readers need to use an IDE/text search to check where this function is being called to make sure it's only used in the effect, otherwise, they could assume it will be used somewhere else, maybe in the render body.